synapse/storage/databases/main/relations.py

```diff
@@ -229,18 +231,42 @@ async def get_relations_for_event(
         if pagination_clause:
             where_clause.append(pagination_clause)
 
-        sql = """
-            SELECT event_id, relation_type, sender, topological_ordering, stream_ordering
-            FROM event_relations
-            INNER JOIN events USING (event_id)
-            WHERE %s
-            ORDER BY topological_ordering %s, stream_ordering %s
-            LIMIT ?
-        """ % (
-            " AND ".join(where_clause),
-            order,
-            order,
-        )
+        # TODO This needs to be a recursive query that maybe still matches some of the times above.
```

Comment:
I'm assuming we need to fix this before we can merge?

Reply:
Errr, I have no idea what this comment is saying. 🤨